ibis/backends/pandas/executor/strings.py

```diff
@@ -0,0 +1,230 @@
+from urllib.parse import parse_qs, urlsplit
+
+import numpy as np
+import pandas as pd
+
+import ibis.expr.operations as ops
+from ibis.backends.pandas.executor.core import execute
+from ibis.backends.pandas.executor.utils import (
+    asframe,
+    columnwise,
+    elementwise,
+    rowwise,
+    serieswise,
+    sql_like_to_regex,
+)
+from ibis.util import any_of
+
+
+def _extract_userinfo_elementwise(x):
+    url_parts = urlsplit(x)
+    username = url_parts.username or ""
+    password = url_parts.password or ""
+    return f"{username}:{password}"
+
+
+def _extract_queryparam_rowwise(row):
+    query = urlsplit(row["arg"]).query
+    param_name = row["key"]
+    if param_name is not None:
+        value = parse_qs(query)[param_name]
+        return value if len(value) > 1 else value[0]
+    else:
+        return query
+
+
+def _substring_rowwise(row):
+    arg, start, length = row["arg"], row["start"], row["length"]
+    if length is None:
+        return arg[start:]
+    else:
+        return arg[start : start + length]
+
+
+def _substring_serieswise(arg, start, length):
+    if length is None:
+        return arg.str[start:]
+    else:
+        return arg.str[start : start + length]
+
+
+_columnwise_functions = {
+    ops.Repeat: lambda df: df["arg"] * df["times"],
+}
+
+_rowwise_functions = {
+    ops.EndsWith: lambda row: row["arg"].endswith(row["end"]),
+    ops.LPad: lambda row: row["arg"].rjust(row["length"], row["pad"]),
+    ops.RegexExtract: lambda row: re.search(row["pattern"], row["arg"]).group(
+        row["index"]
+    ),
+    ops.RegexReplace: lambda row: re.sub(
+        row["pattern"], row["replacement"], row["arg"]
+    ),
+    ops.RegexSearch: lambda row: re.search(row["pattern"], row["arg"]) is not None,
+    ops.RPad: lambda row: row["arg"].ljust(row["length"], row["pad"]),
+    ops.StartsWith: lambda row: row["arg"].startswith(row["start"]),
+    ops.StringContains: lambda row: row["haystack"].contains(row["needle"]),
+    ops.StringFind: lambda row: row["arg"].find(
+        row["substr"], row["start"], row["end"]
+    ),
+    ops.StringReplace: lambda row: row["arg"].replace(
+        row["pattern"], row["replacement"]
+    ),
+    ops.StringSplit: lambda row: row["arg"].split(row["delimiter"]),
+    ops.StrRight: lambda row: row["arg"][-row["nchars"] :],
+    ops.Substring: _substring_rowwise,
+    ops.Translate: lambda row: row["arg"].translate(
+        str.maketrans(row["from_str"], row["to_str"])
+    ),
+    ops.ExtractQuery: _extract_queryparam_rowwise,
+}
+
+_serieswise_functions = {
+    ops.Capitalize: lambda arg: arg.str.capitalize(),
+    ops.EndsWith: lambda arg, end: arg.str.endswith(end),
+    ops.Lowercase: lambda arg: arg.str.lower(),
+    ops.LPad: lambda arg, length, pad: arg.str.rjust(length, fillchar=pad),
+    ops.LStrip: lambda arg: arg.str.lstrip(),
+    # TODO(kszucs): while these are properly working, the pandas implementation
+    # doesn't seem to support POSIX patterns
```

Comment:
This is why the pandas and dask backends depend on the third party `regex` module which does support posix classes.

Reply:
These could have been more performant implementations, but I removed them in favor of the already existing rowwise `regex` functions.